File: src/assets/usage-examples-code/timers/resume_timer__named/resume_timer_named-1-usage-example.cpp
#include <iostream>
#include "splashkit.h"

int main()
{
    std::string timerName = "myTimer";
    timer myTimer = create_timer(timerName);

    for (int i = 4; i > 0; --i)
    {
        std::cout << i << " seconds remaining..." << std::endl;
        delay(1000);
        if (i == 3)
        {
            pause_timer(myTimer);
            std::cout << "Timer paused at 3 seconds." << std::endl;
        }
    }

    // Resume the timer
    std::cout << "Timer resumed..." << std::endl;
    resume_timer(myTimer);

    return 0;
}

File: src/assets/usage-examples-code/timers/resume_timer__named/resume_timer_named-1-usage-example.cs
using System;
using SplashKitSDK;

class Program
{
    static void Main(string[] args)
    {
        string timerName = "myTimer";

        SplashKit.StartTimer(timerName);

        for (int i = 4; i > 0; --i)
        {
            Console.WriteLine(i + " seconds remaining...");
            System.Threading.Thread.Sleep(1000); 
            if (i == 3)
            {
                SplashKit.PauseTimer(timerName);
                Console.WriteLine("Timer paused at 3 seconds.");
            }
        }

        // Resume the timer
        Console.WriteLine("Timer resumed...");
        SplashKit.ResumeTimer(timerName);
    }
}
